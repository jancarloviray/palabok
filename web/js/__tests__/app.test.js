/* eslint-env jest */
import App from '../app'

let app

beforeEach(() => {
  app = new App({ env: 'test' })
  app.start()
})

afterEach(() => {
  app.stop()
})

describe('App', () => {
  test('starts and dies flawlessly', () => {
    expect(1).toEqual(1)
  })

  test('drops the right HTML', () => {
    expect(document.body.innerHTML).toMatchSnapshot()
  })

  test('Empty timer', () => {
    app.store.dispatch({ type: 'router:nav!', to: '/timer' })
  })

  test('Started work timer', () => {
    app.store.dispatch({ type: 'timer:start!', timerType: 'work' })
  })

  test('Started break timer', () => {
    app.store.dispatch({ type: 'timer:start!', timerType: 'break' })
  })

  test('Empty log', () => {
    app.store.dispatch({ type: 'router:nav!', to: '/log' })
  })
})
