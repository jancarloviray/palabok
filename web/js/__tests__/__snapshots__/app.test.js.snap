// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App drops the right HTML 1`] = `"<div id=\\"root\\" class=\\"app-root\\"><noscript data-reactroot=\\"\\"></noscript></div><div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div></div></span></div>"`;
